lib/patcher.js

```diff
@@ -498,10 +499,12 @@ function findChildReprints(newPath, oldPath, reprints) {
   // path.needsParens() && !hasParens(oldNode) check below, the absence of
   // a closing parenthesis after the FunctionExpression would trigger
   // pretty-printing unnecessarily.
-  if (!newPath.canBeFirstInStatement() &&
+  if (Node.check(newNode) &&
+      !newPath.canBeFirstInStatement() &&
       newPath.firstInStatement() &&
-      !hasOpeningParen(oldPath))
+      !hasOpeningParen(oldPath)) {
     return false;
```

Comment:
@xixixao Your test case was triggering this `return false` when `newNode` is the `extra` object, though this logic only applies to `Node` objects.